### DISCUSSION TITLE
[ FF7 ] stutter when use_external_music is set to false

### DISCUSSION THREAD

**Discussion by arthemis841** [julianxhokaxhiu/FFNx#203] [Q&A]
**Describe the bug**
I don't know if it's a bug, but I have noticed that when use_external_music is set to false, the game will have a stutter before starting an encounter and when an encounter is finished. I can see the stutter also looking at the fps, there is a drop from 30 to 24 when starting an encounter. 

If I set use_external_music to true, then the transition when finding an encounter is 100% smooth, with fps staying on locked 30 and then of course transitioning to 15 in battle.

**To Reproduce**
1) use_external_music=false
2) find a random encounter and look for a brief stutter / fps drop

I'm testing without any mod installed, I've the steam version with the lastest 7th heaven installed, and I have manually updated FFNX to 1.8.1.159

**Expected behavior**
I think it shouldn't be stuttering with either setting? I really don't know.

**GPU (please complete the following information):**

- Brand Nvidia RTX 2060
- Driver Version [461.40

**Comment by julianxhokaxhiu** [comment]
Hi,

Thank you very much for the bug reporting. The idea is that it stutters because the original Steam release does not expect to play MIDIs and instead it provides an external audio soundtrack ( OGG files ) to be played when running the game.

Even if you can actually turn it off, it's not expected to be on Steam releases because of this. Also as turning off the flag means allowing the internal engine to do what is expected to ( so we have no more control over it ) minus the fact that midi files are not provided. Because of this a potential solution would be patching the engine and adding an IF check before loading the MIDI file, although I find it redundant as for Steam this is never supposed to happen ( despite you having full control over it ).

I'll leave this bug open for future reference if anyone is willing to tackle this minor issue and create a PR for it.

For the time being consider leaving enabled the external music layer.

Cheers!

**Comment by julianxhokaxhiu** [comment]
So, apparently this is not a bug on its own but probably related to your own environment.

Upon further investigation I forgot that I already tackled this case, by forcing always `use_external_music = true` when you install FFNx as a Steam driver ( see https://github.com/julianxhokaxhiu/FFNx/blob/master/src/common.cpp#L2409 ).

Therefore, if you get a lag when you set this as false, is not related to the flag ( as it will be set anyway to true on boot ).

Because of this I'll close the issue.
